byzcoin/bcadmin/cmd_db.go

```diff
@@ -510,6 +514,23 @@ func dbCheck(c *cli.Context) error {
 				}
 			}
 		}
+
+		indexes := sb.GetFLIndexes()
+		for i, index := range indexes {
+			if index <= last.Index {
+				if i >= len(sb.ForwardLink) {
+					maxLink := len(indexes) - 1
+					for ; indexes[maxLink] > last.Index; maxLink-- {
+					}
+					log.Warnf("%s only %d out of %d forward-links",
+						errStr, i, maxLink+1)
```

Comment:
`i`  is the index of the missing forward link, and `maxLink+1` the index of the highest possible forward link pointing to a block. Therefore, the message should rather be:

```suggestion
						errStr, len(sb.ForwardLink), maxLink+1)
```

Reply:
Nice catch. In fact the forward-links can be even worse than what I wrote here:
Given a block that should have 4 forward-links {0,1,2,3}, it can have only {0,2}, so missing two.

That whole `for`-loop there is bad - I inserted it in the `for`-loop below, I hope that's more clear.